src/tree_cmd.cpp

```diff
@@ -178,6 +184,139 @@ static void PlaceTree(TileIndex tile, uint32_t r)
+			deviation += sin((theta + harmonic.phase) * harmonic.frequency) * harmonic.amplitude;
+		});
+
+		//smooth out changes
+		auto adjustedRadius = (radius / 2.0) + (deviation / 2);
+
+		// add to the final polygon
+		BlobPosition vertex;
+		vertex.x = cos(theta) * adjustedRadius;
+		vertex.y = sin(theta) * adjustedRadius;
+		result.push_back(vertex);
+
+		//proceed to the next segment
+		theta += step;
+	}
+
+	return result;
+}
+
+/**
+ * Creates a random star-shaped[sic] polygon originating from (0, 0).
+ *
+ * @param radius The maximum radius of the blob. May be smaller, but will not be larger.
+ * @param noOfSegments How many segments make up the blob.
+ */
+std::vector<BlobPosition> createRandomStarShapedPolygon(const int radius, const int noOfSegments)
+{
+	std::vector<BlobHarmonic> harmonics;
+
+	// these values are ones i found in my testing that result in suitable- looking polygons that did not self-intersect and fit within a square of radius * radius dimensions.
+
+	harmonics.push_back(BlobHarmonic(radius / 2, Random() / PHASE_DIVISOR, 1));
+	harmonics.push_back(BlobHarmonic(radius / 4, Random() / PHASE_DIVISOR, 2));
+	harmonics.push_back(BlobHarmonic(radius / 8, Random() / PHASE_DIVISOR, 3));
+	harmonics.push_back(BlobHarmonic(radius / 16, Random() / PHASE_DIVISOR, 4));
+
+	return createStarShapedPolygon(radius, harmonics, noOfSegments);
+}
+
+/**
+ * Returns true if the given coordinates lie within a triangle.
+ *
+ * @param x x.
+ * @param y y.
+ * @param triangle the triangle to check against, a std::vector of three BlobPositions.
+ */
+bool pointInTriangle(const int x, const int y, std::vector<BlobPosition> triangle)
+{
+	assert(triangle.size() == 3);
+	const auto vertex0 = triangle.at(0);
+	const auto vertex1 = triangle.at(1);
+	const auto vertex2 = triangle.at(2);
+
+	const int s = ((vertex0.x - vertex2.x) * (y - vertex2.y)) - ((vertex0.y - vertex2.y) * (x - vertex2.x));
+	const int t = ((vertex1.x - vertex0.x) * (y - vertex0.y)) - ((vertex1.y - vertex0.y) * (x - vertex0.x));
+
+	if ((s < 0) != (t < 0) && (s != 0 && t != 0))
+	{
+		return false;
+	}
+
+	const int d = (vertex2.x - vertex1.x) * (y - vertex1.y) - (vertex2.y - vertex1.y) * (x - vertex1.x);
+	return (d < 0) == (s + t <= 0);
+}
+
+/**
+ * Returns true if the given coordinates lie within a star shaped[sic] polygon.
+ * breaks the polygon into a series of triangles around the centre point (0, 0) and then tests the coordinates against each triangle until a match is found [or not].
+ *
+ * There might be a better way to do this.
+ *
+ * @param x x.
+ * @param y y.
+ * @param polygon the polygon to check against, a std::vector of multiple BlobPositions.
+ */
+bool pointInStarShapedPolygon(int x, int y, std::vector<BlobPosition> polygon)
+{
+	for(int i = 0; i < polygon.size(); ++i)
+	{
+		std::vector<BlobPosition> triangle;
+		triangle.push_back(polygon.at(i));
+		triangle.push_back(polygon.at((i + 1) % polygon.size()));
+		triangle.push_back({0, 0});
```

Comment:
Using a vector here instead of just passing 3 `Points` to `pointInTriangle()` is wasteful